bioverse/functions.py

```diff
@@ -177,6 +177,135 @@ def read_stellar_catalog(d, filename=DATA_DIR+'LUVOIR_targets.dat', d_max=30., T
 
     return d
 
+def create_planets_bergsten(d, R_min=1.0, R_max=3.5, P_min=2, P_max=100., seed=42):
+    """ Generates planets with periods and radii according to Bergsten+2022 occurrence rate estimates.
+
+    Parameters
+    ----------
+    d : Table
+        Table containing simulated host stars.
+    R_min : float, optional
+        Minimum planet radius, in Earth units.
+    R_max : float, optional
+        Maximum planet radius, in Earth units.
+    P_min : float, optional
+        Minimum orbital period, in days.
+    P_max : float, optional
+        Maximum orbital period, in days.
+    seed : int or 1-d array_like, optional
+        Seed for numpy's RandomState. Must be convertible to 32 bit unsigned integers.
+
+    Returns
+    -------
+    d : Table
+        Table containing the sample of simulated planets. Replaces the input Table.
+    """
+
+    np.random.seed(seed)
+    
+    # Bins, Parameters, and Values from Bergsten et al. 2022
+    # Stellar Mass Bins
+    massbins = [[0.556,0.815],[0.815,0.909],[0.909,1.008],[1.008,1.16],[1.16,1.629]]
+    # Mass scalings of the Radius Valley
+    Rsplit = [1.8198343592849173,1.92711136271344,1.978919030506143,2.0407383044237983,2.173374790737642]
+
+    # Best-Fit Free Parameters
+    F_0 = [0.8856361259972422,0.6996159608485483,0.6327526676490182,0.6129575597286655,0.5009869054654867]
+    P_break = [14.288973856114884,6.133363167804966,6.918644494007631,12.017636043391624,6.958618711705457]
+    beta1 = [0.15196765221136338,1.1888123910972717,0.911590303062237,0.43848485219532984,1.9016215517122808]
+    beta2 = [-1.153131110847535,-0.6819146491558035,-0.8437029161066527,-1.101432111115821,-0.6907313978319928]
+    P_central = [7.543177275679916,11.257915734447987,12.976947091193168,17.568566667378498,16.567186596597324]
+    s = [1.4622073535853128,2.0225542991906944,2.532054278425091,2.1627543721548346,2.154972607211169]
+    chi1 = [0.7528696587569685,0.73341694903406,0.8304039843638047,0.8254924295852643,0.8725919841057193]
+    chi2 = [0.32550731709760333,0.35832058563162017,0.2560695336448793,0.3084550279549865,0.3926113907097983]
+    # Uncertainty in avg. number of planets per star
+    sigma_F0 = [0.05293837, 0.04864412, 0.04215205, 0.04083521, 0.04121296]
+
+    # pre-computed normalization parameters
+    Cn = [0.04381, 0.05183, 0.06002, 0.05155, 0.05177]
+    
+    ### Need a way to modify for transit mode...?
+#     if transit_mode:
+        
+    # Some empty arrays to temporarily store planet parameters from different stellar mass bins
+    num_planets = np.empty(len(d))
+    master_P, master_R = [],[]
+
+    # Set up probability grid in R and P
+    P,dP = np.linspace(P_min,P_max,1000,retstep=True)
+    R,dR = np.linspace(R_min,R_max,1000,retstep=True)
+    xv, yv = np.meshgrid(P, R)
+    
+    # Generate probability grids in bins of stellar mass
+    for i,_ in enumerate(F_0):
```

Comment:
out of curiosity: this is basically `for i in range(len(F_0)):`, right?

Reply:
Yup! Since only the 'i' index is needed, `range(len(F_0))` is really all that's needed. I just have a nasty habit of using `enumerate` in all my loops...